crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -43,10 +51,35 @@ use tedge_http_ext::HttpResult;
 
 const RETRY_TIMEOUT_SECS: u64 = 20;
 
+pub enum InternalIDUsage {
+    Url,
+    Body,
+}
+
+struct HttpRequestParts {
+    method: Method,
+    uri: Uri,
+    body: String,
+    headers: HeaderMap,
+}
+
+impl HttpRequestParts {
+    fn new(method: Method, uri: Uri, body: String, headers: HeaderMap) -> Self {
+        Self {
+            method,
+            uri,
+            body,
+            headers,
+        }
+    }
+}
+
 pub struct C8YHttpProxyActor {
     end_point: C8yEndPoint,
     child_devices: HashMap<String, String>,
     peers: C8YHttpProxyMessageBox,
+    token: Option<String>,
+    id_usage: InternalIDUsage,
```

Comment:
- Why do you need to know how the internal id is used by a request?
- This seems weird to have this usage attached to the actor while this depends on the request.
- I got a quick look to the code where this used. Please stop doing this. Use method parameters to change the behavior of method!

Reply:
Refactored the code

- Moved the token to the `end_point` struct
- Using the `method`. to update the `body` or `URL` of the request.